pkg/api/testing/fuzzer.go

```diff
@@ -45,6 +45,17 @@ func FuzzerFor(t *testing.T, version string, src rand.Source) *fuzz.Fuzzer {
 		f.RandSource(src)
 	}
 	f.Funcs(
+		func(j *int, c fuzz.Continue) {
+			*j = int(c.Int31())
+		},
+		func(j **int, c fuzz.Continue) {
```

Comment:
int is 64 bit internally, so fuzzInt generates a 64 bit number which won't
cross the boundary..  That won't round trip to the external types.  If /
when we refactor the internal, we would not need to do this.

On Wed, Nov 18, 2015 at 6:09 PM, Daniel Smith <notifications@github.com>
wrote:

> In pkg/api/testing/fuzzer.go
> <https://github.com/kubernetes/kubernetes/pull/17353#discussion_r45275625>
> :
>
> > @@ -45,6 +45,17 @@ func FuzzerFor(t *testing.T, version string, src rand.Source) *fuzz.Fuzzer {
> >  		f.RandSource(src)
> >  	}
> >  	f.Funcs(
> > +		func(j *int, c fuzz.Continue) {
> > +			*j = int(c.Int31())
> > +		},
> > +		func(j **int, c fuzz.Continue) {
>
> I'm surprised you need this function, it should be the default behavior.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17353/files#r45275625>.
>